src/librustc_trans/collector.rs

```diff
@@ -433,6 +434,40 @@ fn check_recursion_limit<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
     (instance.def, recursion_depth)
 }
 
+fn check_type_length_limit<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
```

Comment:
Checking it in other places would basically require us to intern "type sizes" in types, which would bump our memory usage for no very good reason.